modules/common/condition/types.go

```diff
@@ -84,3 +84,9 @@ type conditionGroup struct {
 	severity   Severity
 	conditions Conditions
 }
+
+// Updater - generic type for helping to update conditions
+type Updater interface {
+	Set(c *Condition)
+	MarkTrue(t Type, messageFormat string, messageArgs ...interface{})
+}
```

Comment:
This is the point where my golang knowledge is lacking. Can we somehow make our Condition type somehow closer / aligned with the interface we need here to manipulate that Condition type?